tests/Backend/Snowflake/TypedTableWorkspacesLoadTest.php

```diff
@@ -0,0 +1,811 @@
+            fn(ColumnInterface $c) => $c->getColumnName() === 'Id',
+        );
+        $this->assertCount(1, $idColumn);
+        $idColumn = array_values($idColumn)[0];
+        $this->assertInstanceOf(SnowflakeColumn::class, $idColumn);
+        $this->assertSame('NUMBER', $idColumn->getColumnDefinition()->getType());
+        $this->assertSame('5,1', $idColumn->getColumnDefinition()->getLength());
+
+        // test for invalid columns
+        $options = [
+            'input' => [
+                [
+                    'source' => $tableId,
+                    'destination' => 'languagesIso',
+                    'columns' => [
+                        [
+                            'source' => 'Id',
+                            'type' => 'integer',
+                        ],
+                        [
+                            'source' => 'iso',
+                            'type' => 'varchar',
+                        ],
+                        [
+                            'source' => 'not-a-column',
+                            'type' => 'varchar',
+                        ],
+                    ],
+                ],
+            ],
+        ];
+        $options = InputMappingConverter::convertInputColumnsTypesForBackend(
+            $workspace['connection']['backend'],
+            $options,
+        );
+        try {
+            $workspaces->loadWorkspaceData($workspace['id'], $options);
+            $this->fail('Trying to select a non existent column should fail');
+        } catch (ClientException $e) {
+            $this->assertEquals('storage.tables.nonExistingColumns', $e->getStringCode());
+        }
+    }
+
+    public function testLoadIncrementalWithColumns(): void
+    {
+        $workspaces = new Workspaces($this->workspaceSapiClient);
+        $workspace = $this->initTestWorkspace();
+
+        $backend = WorkspaceBackendFactory::createWorkspaceBackend($workspace);
+
+        $tableId = $this->createTableLanguagesMoreColumns(['Id']);
+
+        // first load
+        $options = [
+            'input' => [
+                [
+                    'source' => $tableId,
+                    'destination' => 'languagesDetails',
+                    'whereColumn' => 'iso',
+                    'whereValues' => ['dd', 'xx'],
+                    'columns' => [
+                        [
+                            'source' => 'Id',
+                            'type' => 'integer',
+                        ],
+                        [
+                            'source' => 'Name',
+                            'type' => 'varchar',
+                        ],
+                    ],
+                ],
+            ],
+        ];
+        $options = InputMappingConverter::convertInputColumnsTypesForBackend(
+            $workspace['connection']['backend'],
+            $options,
+        );
+
+        $workspaces->loadWorkspaceData($workspace['id'], $options);
```

Comment:
tady by bylo cool to assertnout , ze je tech radku mene nez 5. at je to jasny, ze ten incremental fakt frci

Reply:
No to je zajímavé měl by tam byl assert 2 :D tady sem ho odstranil žejo :D https://github.com/keboola/storage-api-php-client/pull/1353#discussion_r1635076206 ale nevím proč byl zakomentovaný :D

Reply:
assert sem přidal

Reply:
jaj, tak to se omlouvam, ja to tehda videl jako proste zakomentovany kod, tak jsem ho chtel pryc...

Reply:
to je v cajku, já sem to prakticky celé copypastnul změnil na typy a úplně sem to tak detailně neprocházal.